clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2505,6 +2505,17 @@ mlir::LogicalResult CIRToLLVMBinOpLowering::matchAndRewrite(
   case cir::BinOpKind::Xor:
     rewriter.replaceOpWithNewOp<mlir::LLVM::XOrOp>(op, llvmTy, lhs, rhs);
     break;
+  case cir::BinOpKind::Max:
+    if (mlir::isa<cir::IntType, mlir::IntegerType>(type)) {
+      auto isUnsigned = mlir::isa<cir::IntType>(type)
+                            ? mlir::cast<cir::IntType>(type).isUnsigned()
+                            : mlir::cast<mlir::IntegerType>(type).isUnsigned();
```

Comment:
Why do we need to check both CIR and LLVM types? If you get the type from the adaptor you only need to check the LLVM one? I see the same happens for `cir::BinOpKind::{Div,Rem}` but since you are here probably clean up that as well?

Reply:
Sure, I was confused why we check mlir::IntegerType too, now it sounds like it might be just historical. I'll clean it up.